crates/router/src/utils.rs

```diff
@@ -70,3 +76,56 @@ pub mod error_parser {
 pub fn generate_id(length: usize, prefix: &str) -> String {
     format!("{}_{}", prefix, nanoid!(length, &consts::ALPHABETS))
 }
+
+pub trait ConnectorResponseExt: Sized {
+    fn get_response(self) -> RouterResult<types::Response>;
+    fn get_error_response(self) -> RouterResult<types::Response>;
+    fn get_response_inner<T: DeserializeOwned, F: FnOnce(Self) -> RouterResult<types::Response>>(
```

Comment:
The function is not required here.

Reply:
But, this function provides a way to access the inner `response` and also parse it to a generic value.

Reply:
But this function provides access to the inner `response` and makes it possible to access it without using `.parse_struct` externally

Reply:
And having a generic function as a argument will also, allow use to parse the inner `error_response` as a struct in a future need arises